aiohttp/client.py

```diff
@@ -220,6 +226,9 @@ def __init__(
         requote_redirect_url: bool = True,
         trace_configs: Optional[List[TraceConfig]] = None,
         read_bufsize: int = 2**16,
+        fallback_charset_resolver: _CharsetResolver = lambda r, b: (
+            chardet.detect(b)["encoding"] or "utf-8"
+        ),
```

Comment:
We're also missing the warning in the second sub-point:
https://github.com/aio-libs/aiohttp/pull/7561#issuecomment-1698287181

Then we can just tweak the docs slightly to match.

Reply:
Yeah, I'll factor it out into an underscored/private method and then pass it

Reply:
@Dreamsorcerer I added a basic deprecation warning. I'm not sure the logic for how to determine what the appropriate `stacklevel` kwarg should be, however. I think `3` might be the "correct" value here.